src/services/Store/StoreProvider.tsx

```diff
@@ -283,6 +286,30 @@ export const StoreProvider: React.FC = ({ children }) => {
     // This interval is used to poll for status of txs.
     const txStatusLookupInterval = setInterval(() => {
       pendingTransactions.forEach((pendingTxReceipt) => {
+        const senderAccount = getAccountByAddressAndNetworkName(
+          pendingTxReceipt.from,
+          pendingTxReceipt.asset.networkId
+        );
+        if (!senderAccount) return;
+
+        const storeAccount = accounts.find((acc) =>
+          isSameAddress(senderAccount.address, acc.address)
+        ) as StoreAccount;
+
+        const txs = getTxsFromAccount([storeAccount]);
+        const overwrittenTx = txs.find(
+          (t) =>
+            t.nonce === pendingTxReceipt.nonce &&
+            t.asset.networkId === pendingTxReceipt.asset.networkId &&
+            t.hash !== pendingTxReceipt.hash &&
+            t.status === ITxStatus.SUCCESS
+        );
+
+        if (overwrittenTx) {
+          removeTxFromAccount(senderAccount, pendingTxReceipt);
+          return;
+        }
```

Comment:
I think this whole bit can be handled in AccountProvider. The AccountProvider already has handling to replace a pending tx with a successful/failed one, so you can probable handle this there as well.

Reply:
I'm not certain this should be the responsibility of the account hook. It will overcomplicate the logic in there. Reached out over Slack.